Microsoft.AVS.Management/Microsoft.AVS.Management.psm1

```diff
@@ -2584,3 +2584,113 @@ Function Get-AVSVSANClusterUNMAPTRIM {
+        [string]
+        $Name,
+        [Parameter(Mandatory = $true)]
+        [bool]
+        $Enable
+    )
+    begin {
+        If ([string]::IsNullOrEmpty($Name)){}
+        Else {
+            $Name = Limit-WildcardsandCodeInjectionCharacters -String $Name
+            $Array = Convert-StringToArray -String $Name
+        }
+        $TagName = "VSAN Encryption in transit"
+        $InfoMessage = "Info - There may be a performance impact when vSAN Encryption in transit is enabled.
+            See url for more information: https://docs.vmware.com/en/VMware-vSphere/7.0/com.vmware.vsphere.vsan.doc/GUID-10099331-92E7-41AF-BCAA-88DB4B4A4B7B.html"
+    }
+    process {
+        If ([string]::IsNullOrEmpty($Array)) {
+            $Clusters = Get-Cluster
+            Foreach ($Cluster in $Clusters) {
+                $vSANConfigView = Get-VsanView -Id VsanVcClusterConfigSystem-vsan-cluster-config-system
+                $vSANReconfigSpec = new-object -type VMware.Vsan.Views.VimVsanReconfigSpec
+                $vSANReconfigSpec.Modify = $true
+                $vSANDataInTransitConfig= new-object -type VMware.Vsan.Views.VsanDataInTransitEncryptionConfig
+                $vSANDataInTransitConfig.Enabled = $Enable
+                $vSANDataInTransitConfig.RekeyInterval = 1440
+                $vSANReconfigSpec.DataInTransitEncryptionConfig = $vSANDataInTransitConfig
+                $task = $vSANConfigView.VsanClusterReconfig($Cluster.ExtensionData.MoRef,$vSANReconfigSpec)
+                Wait-Task -Task (Get-Task -Id $task)
+                If ((Get-Task -Id $task).State -eq "Success"){
+                    Add-AVSTag -Name $TagName -Description $InfoMessage -Entity $Cluster
+                    Write-Host "$($Cluster.Name) set to $Enable for vSAN Encryption in transit"
+                    If ($Enable) {
+                        Write-Information $InfoMessage
+                    }
+                }else {
+                    Write-Error "Failed to set $($Cluster.Name) to $Enable for vSAN Encryption in transit"
+                }
+            }
+        }
+        Else {
+            Foreach ($Entry in $Array) {
+                If ($Cluster = Get-Cluster -name $Entry) {
+                    $vSANConfigView = Get-VsanView -Id VsanVcClusterConfigSystem-vsan-cluster-config-system
+                    $vSANReconfigSpec = New-Object -type VMware.Vsan.Views.VimVsanReconfigSpec
+                    $vSANReconfigSpec.Modify = $true
+                    $vSANDataInTransitConfig= New-Object -type VMware.Vsan.Views.VsanDataInTransitEncryptionConfig
+                    $vSANDataInTransitConfig.Enabled = $Enable
+                    $vSANDataInTransitConfig.RekeyInterval = 1440
+                    $vSANReconfigSpec.DataInTransitEncryptionConfig = $vSANDataInTransitConfig
+                    $task = $vSANConfigView.VsanClusterReconfig($Cluster.ExtensionData.MoRef,$vSANReconfigSpec)
+                    Wait-Task -Task (Get-Task -Id $task)
```

Comment:
how long, roughly, are we waiting? The timeout for the command is enough at 10min?

Reply:
In my testing it's been only about 30 seconds.   But i haven't tested on a large cluster yet.